filters/include/pcl/filters/impl/voxel_grid_covariance.hpp

```diff
@@ -323,8 +323,9 @@ pcl::VoxelGridCovariance<PointT>::applyFilter (PointCloud &output)
         voxel_centroids_leaf_indices_.push_back (static_cast<int> (it->first));
 
       // Single pass covariance calculation
-      leaf.cov_ = (leaf.cov_ - 2 * (pt_sum * leaf.mean_.transpose ())) / leaf.nr_points + leaf.mean_ * leaf.mean_.transpose ();
-      leaf.cov_ *= (leaf.nr_points - 1.0) / leaf.nr_points;
+      const auto additional_num = leaf.nr_points * leaf.mean_ * lead.mean_.transpose() -
```

Comment:
Are you sure that using `auto` here is ok? http://eigen.tuxfamily.org/dox/TopicPitfalls.html

Reply:
Sure. I'll not use auto. BTW, should I rearrange the equation to use 1 transpose or leave it as it is?

Reply:
Hm, since `pt_sum = leaf.nr_points * leaf.mean_` , `additional_num` is basically just `- pt_sum * leaf.mean_.transpose()`

Reply:
PTAL

Reply:
@mvieth